src/main/java/org/folio/edge/sip2/parser/MessageParser.java

```diff
@@ -62,6 +62,17 @@ protected String parseVariableLengthField(char [] messageChars, Field field) {
     return new String(messageChars, startPosition, position - startPosition);
   }
 
+  protected OffsetDateTime parseDateTimeNB(char [] messageChars) {
+    final String dateTimeString = new String(messageChars, position, 18);
+    
+    if (" ".repeat(18).equals(dateTimeString)) {  // return null for 18 space nb due date
```

Comment:
Forgot about `String.repeat(count)` being added in Java 11. Interesting use. Maybe we should make it a constant in this class though? I don't see it changing ever, so I think it would be a good constant candidate, otherwise it is kind of a magic string, though the comment helps. Also, I took a look at the `repeat` implementation and, for 1 char `String` objects, like this, it uses `Arrays.fill()`, which is fine, but we don't need to do that every time we parse a message. I think something like `private static final String MISSING_NB_DUE_DATE = " ".repeat(18);` makes sense to me.

Anyway, this is very minor, so feel free to ignore.